lib/xcpretty/syntax.rb

```diff
@@ -1,44 +1,39 @@
+begin
+  require 'rouge'
+rescue LoadError
+  Rouge = nil
+end
+
 require 'xcpretty/snippet'
 
 module XCPretty
   class Syntax
-    def self.register_filetype(type, extensions)
-      @filetypes ||= {}
-      extensions.each { |ext| @filetypes[ext] = type }
-    end
-
-    register_filetype 'c++',    ['.cpp', '.hpp', '.c++', '.cxx', '.cc']
-    register_filetype 'objc',   ['.m', '.h']
-    register_filetype 'objc++', ['.mm', '.hh']
-    register_filetype 'swift',  ['.swift']
-    register_filetype 'dylan',  ['.dyl', '.dylan']
-    register_filetype 'ruby',   ['.ruby', '.rb']
+    def self.highlight(snippet)
+      return snippet.contents unless Rouge
 
-    def self.highlight(snippet, options = '')
-      if Pygments.available?
-        language = file_language(File.basename(snippet.file_path))
-        Pygments.pygmentize(snippet.contents, language, options)
+      if snippet.file_path.include?(':')
+        filename = snippet.file_path.rpartition(':').first
       else
-        snippet.contents
+        filename = snippet.file_path
       end
-    end
-
-    private
 
-    def self.file_language(filename)
-      ext = File.extname(filename)
-      @filetypes[ext] || 'objc'
-    end
-  end
+      options = {
+        :filename => File.basename(filename),
+        :source => snippet.contents,
+      }
+      lexer = Rouge::Lexer.guesses(options).first
```

Comment:
It's not as accurate since certain pieces of code might have false positives against other languages. Especially when using BDD style frameworks.

I don't think performance is much of a problem since this code path is *only* called when there is a failing test (and there shouldn't be many of these ;)). This is also how it worked before so it was acceptable beforehand.